packages/victory-histogram/src/helper-methods.js

```diff
@@ -0,0 +1,303 @@
+import { assign, isNil } from "lodash";
+import { Helpers, LabelHelpers, Data, Domain, Scale } from "../../victory-core/src";
+import { getBarPosition } from "../../victory-bar/src/helper-methods";
+import isEqual from "react-fast-compare";
+import * as d3Array from "d3-array";
+import * as d3Scale from "d3-scale";
+
+const addDays = (date, days) => {
+  const result = new Date(date);
+  result.setDate(result.getDate() + days);
+  return result;
+};
+
+const addMonths = (date, months) => {
+  const result = new Date(date);
+  result.setMonth(result.getMonth() + months);
+  return result;
+};
+
+const addYears = (date, years) => {
+  const result = new Date(date);
+  result.setFullYear(result.getFullYear() + years);
+  return result;
+};
+
+const getNextDate = (date, strategy) => {
+  if (strategy === "day") {
+    return addDays(date, 1);
+  } else if (strategy === "month") {
+    return addMonths(date, 1);
+  }
+
+  return addYears(date, 1);
+};
+
+const resetDate = (date, strategy) => {
+  const newDate = new Date(date);
+  if (strategy === "day") {
+    newDate.setHours(0, 0, 0, 0);
+  } else if (strategy === "month") {
+    newDate.setHours(0, 0, 0, 0);
+    newDate.setDate(1);
+  } else {
+    newDate.setHours(0, 0, 0, 0);
+    newDate.setDate(1);
+    newDate.setMonth(0);
+  }
+
+  return newDate;
+};
+
+const cacheLastValue = (func) => {
+  let called = false;
+  let lastArgs;
+  let lastReturnVal;
+
+  return (...args) => {
+    if (called && isEqual(lastArgs, args)) {
+      return lastReturnVal;
+    }
+
+    const value = func(...args);
+
+    called = true;
+    lastReturnVal = value;
+    lastArgs = args;
+
+    return value;
+  };
+};
+
+const getBinningFunc = ({ data, x, bins }) => {
+  const xAccessor = Helpers.createAccessor(x || "x");
+  const dataIsDates = data.some((datum) => xAccessor(datum) instanceof Date);
+  const bin = d3Array.bin().value(xAccessor);
+  const niceScale = d3Scale
+    .scaleLinear()
```

Comment:
I think this should use the x domain / scale passed in or calculated from props

Reply:
Actually, nevermind. I understand that this domain is only used for calculating the bin edges, so it makes sense for it to be scoped to just the data / bins of this component.

Reply:
If you allow time scales as well as linear scales, can you get around needing to use "day" "month" "year" bins?